src/settingsview.cpp

```diff
@@ -184,3 +200,13 @@ void SettingsView::onReopenTabChanged(bool reopen)
 {
     ui->reopenTabToggle->setChecked(reopen);
 }
+
+QString SettingsView::formatDownloadDir(const QString& input) {
+    const int maxLength = 40;
+    if (input.length() > maxLength) {
+        QString suffix = input.right(maxLength);
+        int directoryIndex = suffix.indexOf('/');
+        return "..." + suffix.mid(directoryIndex != -1 ? directoryIndex : 0);
+    }
+    return input;
+}
```

Comment:
This doesn't need to be a member function of `SettingsView`. Please make it a free function in an unnamed namespace (and don't forget to delete its declaration from the *.h file).

Reply:
Is it ok to keep in the same file and just remove the namespace?

Reply:
Yes